src/pages/products-detail/index.tsx

```diff
@@ -20,17 +18,30 @@ import SheetRenderer from './components/sheet-renderer'
 import { reviewData as const_review_data, mockCard } from './mockData'
 import * as S from './ProductsDetail.style'
 
+import { setProductDetail } from '@/store/product-slice/product-slice'
+import { sheet } from '@/store/sheet-slice/sheet-slice'
+import type { ISheetSliceState } from '@/store/sheet-slice/sheet-slice.type'
+import { RootState } from '@/store/store'
+import { IProductCardData } from '@components/product-card/ProductCard.type'
 import type { ISheetComponents } from './ProductsDetail.type'
 
-import { IProductCardData } from '@components/product-card/ProductCard.type.ts'
-
 function ProductsDetail() {
   const { productId } = useParams()
+  const dispatch = useDispatch()
+
   const { data: productDetailQuery, isSuccess: isProductDetailSuccess } =
     useQuery({
       queryKey: ['products-detail', productId],
       queryFn: ({ queryKey }) => getProductDetail(Number(queryKey[1])),
     })
+
+  useEffect(() => {
+    if (productDetailQuery?.data) {
+      dispatch(setProductDetail(productDetailQuery.data))
+    }
+  }, [productDetailQuery, dispatch])
+
+  const productDetail = useSelector((state: RootState) => state.product.detail)
```

Comment:
P3: 먼저, dispatch는 불변함수 이므로, 제거해도 될 거 같습니다.

+ 추가로 저는 reactQuery를 사용한다면 useEffect 사용보다는 reacQuery에서 제공하는 콜백을 사용하는 것이 좋아 보입니다. 
useEffect는 의존성에 의해서 함수가 호출되므로 불필요한 호출이 발생한다고 생각해서, reacQuery의 onSuccess 콜백을 사용한다면 데이터를 가져오는데 성공했을 때만 호출되므로 이 방법을 제안드립니다! 
```suggestion
const { data: productDetailQuery, isSuccess: isProductDetailSuccess } = useQuery({
  queryKey: ['products-detail', productId],
  queryFn: ({ queryKey }) => getProductDetail(Number(queryKey[1])),
  onSuccess: (data) => {
    dispatch(setProductDetail(data));
  }
});
```